core/tests/guild/test_guild_message.py

```diff
@@ -0,0 +1,271 @@
+import importlib
+import time
+
+import pytest
+
+from rustic_ai.core import (
+    AgentTag,
+    GuildTopics,
+    Message,
+    MessagingConfig,
+    Priority,
+)
+from rustic_ai.core.agents.system.models import (
+    UserAgentCreationRequest,
+    UserAgentCreationResponse,
+)
+from rustic_ai.core.agents.testutils import ProbeAgent
+from rustic_ai.core.agents.utils import UserProxyAgent
+from rustic_ai.core.guild.builders import AgentBuilder, GuildBuilder
+from rustic_ai.core.messaging import MessageConstants
+from rustic_ai.core.utils import GemstoneGenerator
+
+from core.tests.guild.simple_agent import (
+    DummyMessage,
+)
+
+
+class TestGuildMessages:
+
+    @pytest.fixture(
+        scope="function",  # Changed from class to function to access messaging_server
+        params=[
+            pytest.param(
+                "InMemoryMessagingBackend",
+                id="InMemoryMessagingBackend",
+            ),
+            pytest.param(
+                "EmbeddedMessagingBackend",
+                id="EmbeddedMessagingBackend",
+            ),
+        ],
+    )
+    def messaging(self, request, messaging_server) -> MessagingConfig:
+        backend_type = request.param
+
+        if backend_type == "InMemoryMessagingBackend":
+            return MessagingConfig(
+                backend_module="rustic_ai.core.messaging.backend",
+                backend_class="InMemoryMessagingBackend",
+                backend_config={},
+            )
+        elif backend_type == "EmbeddedMessagingBackend":
+            # Use the shared messaging server from conftest.py
+            server, port = messaging_server
+            return MessagingConfig(
+                backend_module="rustic_ai.core.messaging.backend.embedded_backend",
+                backend_class="EmbeddedMessagingBackend",
+                backend_config={"auto_start_server": False, "port": port},
+            )
+        else:
+            raise ValueError(f"Unknown backend type: {backend_type}")
+
+    def test_message_broadcast(self, messaging: MessagingConfig, database, org_id):
```

Comment:
this test was just moved here as it was not testing builder functionality but messaging